cpp/src/arrow/util/tracing_internal.h

```diff
@@ -146,6 +210,19 @@ opentelemetry::trace::StartSpanOptions SpanOptionsWithParent(
         return st;                                                                \
       })
 
+#define GET_CURRENT_SPAN(span) \
+  auto span = ::arrow::internal::tracing::GetTracer()->GetCurrentSpan()
+
+#define SET_SPAN_SCOPE(scope, span) \
+  auto scope = ::arrow::internal::tracing::GetTracer()->WithActiveSpan(span)
+
+#define TIE_SPAN_TO_GENERATOR(generator) \
+        generator = ::arrow::internal::tracing::TieSpanToAsyncGenerator(generator)
+
+#define PROPAGATE_SPAN_TO_GENERATOR(generator) \
+        generator = \
+            ::arrow::internal::tracing::PropagateSpanThroughAsyncGenerator(generator)
+
```

Comment:
These macros seem more like aliases.  I'm not sure they help more than they obfuscate.

Reply:
Their purpose is to have blank versions when compiling without opentelemetry
```
#ifdef ARROW_WITH_OPENTELEMETRY
#define GET_CURRENT_SPAN(span) \
  auto span = ::arrow::internal::tracing::GetTracer()->GetCurrentSpan()
...
#else
#define START_SPAN(target_span, ...)
```
So if I remove them, I need to guard all of the calls to these functions with `#ifdef ARROW_WITH_OPENTELEMETRY`